### DISCUSSION TITLE
AMD dependency discussion thread

### DISCUSSION THREAD

**Discussion by Spencer-Dawson** [oobabooga/text-generation-webui#1258] [Ideas]
text-generation-webui works fine on AMD hw, but some of the dependencies do not. 

I'd like to see better support for that and this community seems like a good place to talk about that, but it's outside the scope of this particular project to "fix" other projects hense the thread and not a bug. 

In particular two requirements come to mind as not playing nice with ROCm; bitsandbytes and GPTQ

I've been using this version of bitsandbytesrocm https://git.ecker.tech/mrq/bitsandbytes-rocm personally since the other versions I've found wouldn't compile.

I've also noticed just today that this repo owner has his own amd install instructions, a bitsandbytes-rocm fork, and has had been active recently. I'm not endorsing this repo as I haven't tried it, but I'll probably try it soon. https://github.com/agrocylo/bitsandbytes-rocm

As for GPTQ I haven't had much time to mess with it yet, but I've gotten errors trying to run models reported to work for NVIDIA cards and I've seen complaints about amd specific performance issues.

So I guess the reason for this thread is this. AMD GPU owners what are your dependency pain points, and suggestions for fixing them?

**Comment by shaunclimbon** [comment]
There's a recent blog post that touches on GPTQ and ROCm, but haven't tried it myself.
- https://huggingface.co/blog/chatbot-amd-gpu

**Comment by evshiron** [comment]
A guide for RX 7900 XTX with BitsAndBytes and GPTQ can be found here:

* https://are-we-gfx1100-yet.github.io/post/text-gen-webui/

Which is adapted from https://rentry.org/eq3hg.

https://github.com/agrocylo/bitsandbytes-rocm is promising and definitely works, however it doesn't support 4-bit due to it's version, and the performance is not good.

https://github.com/WapaMario63/GPTQ-for-LLaMa-ROCm is working well in 4-bit inference, but I am getting errors when quantizing.

`triton` doesn't seem to work with my RX 7900 XTX at all. I have a request here. I would like users of gfx1030 to run unit tests on [this branch of triton](https://github.com/ROCmSoftwarePlatform/triton/tree/pytorch-triton-rocm-v2.1.0) in order to compare results with gfx1100:

```bash
# clone
git clone -b pytorch-triton-rocm-v2.1.0 https://github.com/ROCmSoftwarePlatform/triton
cd triton

# activate venv
python3 -m venv venv
source venv/bin/activate

cd python
pip install cmake

# build triton and install dev dependencies
pip install -e .
pip install -e '.[tests]'

# do things like replace torch here

# run unit tests
pytest -vs test/unit/ 2>&1 | tee test.log
```
